leaf-ui/rappid-extensions/BackboneModelsIntention.js

```diff
@@ -39,17 +38,526 @@ var EvolvingFunctionBBM = Backbone.Model.extend({
        
     },
     
-    //TODO: I think could have some helper functions for the constructor. 
+    //TODO: I think could have some helper functions for the constructor.
+    
+    // TODO update all of the function descriptions
+
+    /**
+     * Returns the 4 digit representation for this
+     * EvolvingFunction's last function segment's
+     * marked value value
+     */
+    getLastMarkedVal: function() {
+        var len = this.functionSegList.length;
+        if (len > 0) {
+            return this.functionSegList[len - 1].get('refEvidencePair');
+        }
+        // Original Function
+        // if (len > 0) {
+        //     return this.functionSegList[len - 1].funcX;
+        // }
+    },
+
+    /**
+     * Returns the 4 digit representation for this
+     * EvolvingFunction's second last function segment's
+     * marked value value. If there is no second last function
+     * segment, this function returns the 4 digit representation of 
+     * the only function segment's marked value
+     */
+    getSecondLastMarkedVal: function() {
```

Comment:
Does it make sense to combine this and getLastMarkedVal and just pass in how far back you want to go as an integer parameter?

Reply:
getIndexLastMarkedVal: function(index) { 
        var len = this.functionSegList.length; 
        if (len > 0 && index <= len) { 
            return this.functionSegList[len - index].get('refEvidencePair'); 
        }
    }

Reply:
Maybe getIndexVal?

Reply:
> Maybe getIndexVal?

I changed it to getNthLastMarkedVal with n as the parameter instead, if that works.